### PR TITLE
provide macos universal2 wheel

### PR DESCRIPTION
**Pull request by daquexian**
### Description
Fixes https://github.com/onnx/onnx/issues/4436

It is easier than I thought to build macos universal2 binaries. A proper `CMAKE_OSX_ARCHITECTURES` is all we need 😂

I have tested the universal2 wheel on my m1 machine.

### CODE REVIEW

**Comment by jcwchen** [comment]
Thank you for the contribution! Please allow me to reopen this PR to trigger release CIs

File: .github/workflows/release_mac.yml [hunk 1]
```diff
@@ -52,14 +53,19 @@ jobs:
...
         source workflow_scripts/protobuf/build_protobuf_unix.sh $NUM_CORES $(pwd)/protobuf/protobuf_install
         export CMAKE_ARGS="-DONNX_USE_LITE_PROTO=ON"
         # Currently GitHub Action agent is using macos-11, use -p to force change the final MACOSX_DEPLOYMENT_TARGET
         # Change -p if MACOSX_DEPLOYMENT_TARGET is different
         if [ '${{ github.event_name }}' == 'schedule' ]; then
-          python setup.py bdist_wheel -p macosx_10_12_x86_64 --weekly_build
+          python setup.py bdist_wheel -p macosx_10_12_${{ matrix.target-architecture }} --weekly_build
         else
-          python setup.py bdist_wheel -p macosx_10_12_x86_64
+          python setup.py bdist_wheel -p macosx_10_12_${{ matrix.target-architecture }}
```
Comment (jcwchen) [inline .github/workflows/release_mac.yml:68]: nit: not sure whether we want to use `macosx_10_12_` for universal2 as well. Probably we can assume only 10.15+ users will try universal2 wheel.

**Review by jcwchen — APPROVED** [review]
Overall looks good to me. I also tested this universal2 wheel on my Mac (non M1/M2) and it worked well.

File: .github/workflows/release_mac.yml (hunk 1 above)
Comment (daquexian) [reply .github/workflows/release_mac.yml:68]: Thanks for your review! It is what I ignored.

What's the reason for choosing 10.15? The minimum macos version that can run on apple silicon chips seems 11. Maybe we can use `macosx_11_0_`.

Comment (jcwchen) [reply .github/workflows/release_mac.yml:68]: Good point. `macosx_11_0_` looks better. Thanks!

Comment (daquexian) [reply .github/workflows/release_mac.yml]: Updated. Waiting for CI :)

Comment (daquexian) [reply .github/workflows/release_mac.yml:68]: @jcwchen I found that python3.7 is incompatible with `macosx_11_0_`.

```bash
$ python3.7 -m pip install onnx-1.12.0-cp37-cp37m-macosx_11_0_x86_64.whl
ERROR: onnx-1.12.0-cp37-cp37m-macosx_11_0_x86_64.whl is not a supported wheel on this platform.
$ python3.7 -m pip install onnx-1.12.0-cp37-cp37m-macosx_10_15_x86_64.whl
Processing ...
$ python3.8 -m pip install onnx-1.12.0-cp38-cp38-macosx_11_0_x86_64.whl
Processing ...
```

So I prefer keeping the versions of x86_64 and universal2 wheels the same. It is also what [numpy](https://pypi.org/project/numpy/1.21.0/#files) does:

image

**Pull request merged**